src/TestableHttpClient/TestableHttpMessageHandler.cs

```diff
@@ -1,55 +1,81 @@
 ﻿using System.Collections.Concurrent;
 
+using TestableHttpClient.Response;
+
 namespace TestableHttpClient;
 /// <summary>
 /// A testable HTTP message handler that captures all requests and always returns the same response.
 /// </summary>
 public class TestableHttpMessageHandler : HttpMessageHandler
 {
     private readonly ConcurrentQueue<HttpRequestMessage> httpRequestMessages = new ConcurrentQueue<HttpRequestMessage>();
-    private Func<HttpRequestMessage, HttpResponseMessage> responseFactory = DefaultResponseFactory;
-
-    private static HttpResponseMessage DefaultResponseFactory(HttpRequestMessage requestMessage) => new HttpResponseMessage(HttpStatusCode.OK)
-    {
-        RequestMessage = requestMessage,
-        Content = new StringContent(string.Empty)
-    };
+    private IResponse response = new StatusCodeResponse(HttpStatusCode.OK);
+    private Func<HttpRequestMessage, HttpResponseMessage>? responseFactory;
 
     /// <summary>
     /// Gets the collection of captured requests made using this HttpMessageHandler.
     /// </summary>
     public IEnumerable<HttpRequestMessage> Requests => httpRequestMessages;
 
-    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
     {
         httpRequestMessages.Enqueue(request);
 
-        var response = responseFactory(request);
+        HttpResponseMessage responseMessage;
+        if (responseFactory is not null)
+        {
+            responseMessage = responseFactory(request);
+        }
+        else
+        {
+            responseMessage = new();
+            HttpResponseContext context = new(request, responseMessage);
+            await response.ExecuteAsync(context, cancellationToken).ConfigureAwait(false);
+        }
 
-        if (response is TimeoutHttpResponseMessage)
+        if (responseMessage.RequestMessage is null)
         {
-            var cancelationSource = cancellationToken.GetSource();
+            responseMessage.RequestMessage = request;
+        }
 
-            if (cancelationSource is not null)
-            {
-                cancelationSource.Cancel(false);
-            }
-            return Task.FromCanceled<HttpResponseMessage>(cancellationToken);
+        if (responseMessage.Content is null)
+        {
+            responseMessage.Content = new StringContent("");
```

Comment:
Check whether or not this is called on .NET6, I suspect not.
If not, add compiler conditionals.